crates/accelerate/src/unitary_compose.rs

```diff
@@ -153,33 +152,8 @@ fn _einsum_matmul_index(qubits: &[u32], num_qubits: usize) -> String {
     )
 }
 
-pub fn gate_fidelity(
-    left: &ArrayView2<Complex64>,
-    right: &ArrayView2<Complex64>,
-    qargs: Option<&[Qubit]>,
-) -> (f64, f64) {
-    let dim = left.nrows(); // == left.ncols() == right.nrows() == right.ncols()
-                            // let trace = left.t().mapv(|el| el.conj()).dot(right).diag().sum();
-
-    let left = left.t().mapv(|el| el.conj());
-    let product = match dim {
-        2 => mm1q(&left.view(), right),
-        4 => mm2q(&left.view(), right, qargs.unwrap_or(&[Qubit(0), Qubit(1)])),
-        _ => left.dot(right),
-    };
-    let trace = product.diag().sum();
-
-    let dim = dim as f64;
-    let normalized_trace = trace / dim;
-    let phase = normalized_trace.arg(); // compute phase difference
-
-    let process_fidelity = normalized_trace.abs().powi(2);
-    let gate_fidelity = (dim * process_fidelity + 1.) / (dim + 1.);
-    (gate_fidelity, phase)
-}
-
-fn mm1q(left: &ArrayView2<Complex64>, right: &ArrayView2<Complex64>) -> Array2<Complex64> {
-    let mut out = arr2(&[[C_ZERO, C_ZERO], [C_ZERO, C_ZERO]]);
+pub fn matmul_1q(left: &ArrayView2<Complex64>, right: &ArrayView2<Complex64>) -> Array2<Complex64> {
```

Comment:
```suggestion
#[inline]
pub fn matmul_1q(left: &ArrayView2<Complex64>, right: &ArrayView2<Complex64>) -> Array2<Complex64> {
```

Reply:
Actually you can ignore these `inline` comments` These are all using heap allocated view objects so there is no real benefit here probably. If this was nalgebra or fixed sized arrays it would be more important. It was just a gut reaction when seeing the `pub` and how these functions were used.